torchgeo/samplers/utils.py

```diff
@@ -47,11 +53,17 @@ def get_random_bounding_box(
 
     width = (bounds.maxx - bounds.minx - t_size[1]) // res
     minx = random.randrange(int(width)) * res + bounds.minx
-    maxx = minx + t_size[1]
+    if sample_mode == SIZE_IN_CRS_UNITS:
+        maxx = minx + t_size[1]
+    elif sample_mode == SIZE_IN_PIXELS:
+        maxx = minx + t_size[1] * res
```

Comment:
```suggestion
    maxx = minx + t_size[1]
    if sample_mode == SIZE_IN_PIXELS:
        maxx = maxx * res
```
Feel free to ignore if its better to be explicit